sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -0,0 +1,287 @@
+
+  def run(): Unit = {
+    logWarning(s"Waiting for connection from Python worker")
+    val channel = serverChannel.accept()
+    logWarning(s"listening on channel - ${channel.getLocalAddress}")
+
+    inputStream = new DataInputStream(
+      Channels.newInputStream(channel))
+    outputStream = new DataOutputStream(
+      Channels.newOutputStream(channel)
+    )
+
+    while (channel.isConnected &&
+      statefulProcessorHandle.getHandleState != StatefulProcessorHandleState.CLOSED) {
+
+      try {
+        logWarning(s"reading the version")
+        val version = inputStream.readInt()
+
+        if (version != -1) {
+          logWarning(s"version = ${version}")
+          assert(version == 0)
+          val messageLen = inputStream.readInt()
+          logWarning(s"parsing a message of ${messageLen} bytes")
+
+          val messageBytes = new Array[Byte](messageLen)
+          inputStream.read(messageBytes)
+          logWarning(s"read bytes = ${messageBytes.mkString("Array(", ", ", ")")}")
+
+          val message = StateRequest.parseFrom(ByteString.copyFrom(messageBytes))
+
+          logWarning(s"read message = $message")
+          handleRequest(message)
+          logWarning(s"flush output stream")
+
+          outputStream.flush()
+        }
+      } catch {
+        case _: EOFException =>
+          logWarning(s"No more data to read from the socket")
+          statefulProcessorHandle.setHandleState(StatefulProcessorHandleState.CLOSED)
+          return
+        case e: Exception =>
+          logWarning(s"Error reading message: ${e.getMessage}")
+          sendResponse(1, e.getMessage)
+          outputStream.flush()
+          statefulProcessorHandle.setHandleState(StatefulProcessorHandleState.CLOSED)
+          return
+      }
+    }
+    logWarning(s"done from the state server thread")
+  }
+
+  private def handleRequest(message: StateRequest): Unit = {
+    if (message.getMethodCase == StateRequest.MethodCase.STATEFULPROCESSORCALL) {
+      val statefulProcessorHandleRequest = message.getStatefulProcessorCall
+      if (statefulProcessorHandleRequest.getMethodCase ==
+        StatefulProcessorCall.MethodCase.SETHANDLESTATE) {
+        val requestedState = statefulProcessorHandleRequest.getSetHandleState.getState
+        requestedState match {
+          case HandleState.CREATED =>
+            logWarning(s"set handle state to Created")
+            statefulProcessorHandle.setHandleState(StatefulProcessorHandleState.CREATED)
+          case HandleState.INITIALIZED =>
+            logWarning(s"set handle state to Initialized")
+            statefulProcessorHandle.setHandleState(StatefulProcessorHandleState.INITIALIZED)
+          case HandleState.CLOSED =>
+            logWarning(s"set handle state to Closed")
+            statefulProcessorHandle.setHandleState(StatefulProcessorHandleState.CLOSED)
+          case _ =>
+        }
+        sendResponse(0)
+      } else if (statefulProcessorHandleRequest.getMethodCase ==
```

Comment:
This method is becoming huge. Can we split each case/msg into a separate function maybe ?